meta-zarhus-webkit/recipes-networking/cog/cog_%.bbappend

```
@@ -0,0 +1,31 @@
+FILESEXTRAPATHS:prepend := "${THISDIR}/${PN}:"
+
+DEPENDS:append:${PN}-autostart = " \
+    nginx \
+    "
+
+PROVIDES:append:${PN} = " \
+    ${PN}-autostart \
+    "
+
+SRC_URI:append:${PN}-autostart = " \
+    file://cog-autostart.service \
+    "
+
+PACKAGES += " \
+    ${PN}-autostart \
+    "
+
+inherit systemd
+
+COG_WIDTH ?= "1920"
+COG_HEIGHT ?= "1080"
+
+SYSTEMD_AUTO_ENABLE:${PN}-autostart = "enable"
+SYSTEMD_SERVICE:${PN}-autostart = "cog-autostart.service"
+
+do_install:append:${PN}-autostart() {
+    sed -i 's/COG_PLATFORM_WL_VIEW_WIDTH="[0-9]*"/COG_PLATFORM_WL_VIEW_WIDTH="${COG_WIDTH}"/' ${WORKDIR}/cog-autostart.service
+    sed -i 's/COG_PLATFORM_WL_VIEW_HEIGHT="[0-9]*"/COG_PLATFORM_WL_VIEW_HEIGHT="${COG_HEIGHT}"/' ${WORKDIR}/cog-autostart.service
+    install -D -m 0644 "${WORKDIR}/cog-autostart.service" "${D}${systemd_system_unitdir}/cog-autostart.service"
```

```suggestion
    install -D -m 0644 "${WORKDIR}/cog-autostart.service" "${D}${systemd_system_unitdir}/cog-autostart.service"
    sed -i 's/COG_PLATFORM_WL_VIEW_WIDTH="[0-9]*"/COG_PLATFORM_WL_VIEW_WIDTH="${COG_WIDTH}"/' ${D}${systemd_system_unitdir}/cog-autostart.service
    sed -i 's/COG_PLATFORM_WL_VIEW_HEIGHT="[0-9]*"/COG_PLATFORM_WL_VIEW_HEIGHT="${COG_HEIGHT}"/' ${D}${systemd_system_unitdir}/cog-autostart.service
```

It's probably better to modify file in `${D}`. Couple examples: https://git.yoctoproject.org/poky/tree/meta/recipes-graphics/wayland/weston-init.bb

That way each time you/bitbake does `do_install` you start from fresh state. Otherwise I think this file would get replaced with original only in `do_unpack` task.

---

Fixed here https://github.com/zarhus/meta-zarhus/compare/99fb8646324027dfbb120d6f77d4cc48ea5895cd..cc32eea0f94d434f527cb4faed54b152b556cc6e.